File: platform/android/MapLibreAndroid/src/opengl/java/org/maplibre/android/maps/renderer/surfaceview/GLSurfaceViewMapRenderer.java
package org.maplibre.android.maps.renderer.surfaceview;

import android.content.Context;
import androidx.annotation.NonNull;

import org.maplibre.android.maps.renderer.egl.EGLConfigChooser;
import org.maplibre.android.maps.renderer.egl.EGLContextFactory;
import org.maplibre.android.maps.renderer.egl.EGLWindowSurfaceFactory;
import org.maplibre.android.maps.renderer.MapRenderer;

public class GLSurfaceViewMapRenderer extends SurfaceViewMapRenderer {

  public GLSurfaceViewMapRenderer(Context context,
                                @NonNull MapLibreGLSurfaceView surfaceView,
                                String localIdeographFontFamily) {
    super(context, surfaceView, localIdeographFontFamily);

    surfaceView.setEGLContextFactory(new EGLContextFactory());
    surfaceView.setEGLWindowSurfaceFactory(new EGLWindowSurfaceFactory());
    surfaceView.setEGLConfigChooser(new EGLConfigChooser());
    surfaceView.setRenderer(this);
    surfaceView.setRenderingRefreshMode(MapRenderer.RenderingRefreshMode.WHEN_DIRTY);
    surfaceView.setPreserveEGLContextOnPause(true);
  }
}

File: platform/android/MapLibreAndroid/src/sharedRenderer/opengl/java/org/maplibre/android/maps/renderer/surfaceview/GLSurfaceViewMapRenderer.kt
package org.maplibre.android.maps.renderer.surfaceview

import android.content.Context
import org.maplibre.android.maps.renderer.MapRenderer
import org.maplibre.android.maps.renderer.egl.EGLConfigChooser
import org.maplibre.android.maps.renderer.egl.EGLContextFactory
import org.maplibre.android.maps.renderer.egl.EGLWindowSurfaceFactory

open class GLSurfaceViewMapRenderer(
    context: Context,
    surfaceView: MapLibreGLSurfaceView,
    localIdeographFontFamily: String?,
) : SurfaceViewMapRenderer(context, surfaceView, localIdeographFontFamily) {
    init {
        surfaceView.setEGLContextFactory(EGLContextFactory())
        surfaceView.setEGLWindowSurfaceFactory(EGLWindowSurfaceFactory())
        surfaceView.setEGLConfigChooser(EGLConfigChooser())
        surfaceView.setRenderer(this)
        surfaceView.setRenderingRefreshMode(MapRenderer.RenderingRefreshMode.WHEN_DIRTY)
        surfaceView.preserveEGLContextOnPause = true
    }
}
